python/pyspark/sql/tests.py

```diff
@@ -2038,6 +2038,61 @@ def test_BinaryType_serialization(self):
         df = self.spark.createDataFrame(data, schema=schema)
         df.collect()
 
+    def test_bucketed_write(self):
+        data = [
+            (1, "foo", 3.0), (2, "foo", 5.0),
+            (3, "bar", -1.0), (4, "bar", 6.0),
+        ]
+        df = self.spark.createDataFrame(data, ["x", "y", "z"])
+
+        # Test write with one bucketing column
+        df.write.bucketBy(3, "x").mode("overwrite").saveAsTable("pyspark_bucket")
+        self.assertEqual(
+            len([c for c in self.spark.catalog.listColumns("pyspark_bucket")
+                 if c.name == "x" and c.isBucket]),
```

Comment:
We can simplify this to 
  
    catalog =  self.spark.catalog

    sum(c.name == "x" and c.isBucket for c in catalog.listColumns("pyspark_bucket"))

f you think this is more readable but i am not convinced that it makes sense to use a separate variable here. We have a few tests like this, don't care about the sequence itself, and I think it would only pollute the scope. But if you have strong feelings about I am happy to adjust it.

Regarding the comment style... Right now (excluding `bucket` by and `sortBy`) we have 

- 23 docstrings with:

        """....
        """

- 7 docstrings:

        """"
        ....
        """"

in `readwriter`.  As you said both are valid, but if we want to keep only one convention it would be a good idea to adjust a whole module.